contracts/evmscript/EVMScriptRunner.sol

```diff
@@ -24,7 +26,9 @@ contract EVMScriptRunner is AppStorage, EVMScriptRegistryConstants {
 
         require(executorAddr.delegatecall(sig, calldataArgs));
 
-        return returnedDataDecoded();
+        output = returnedDataDecoded();
+
+        ScriptResult(_script, _input, output);
```

Comment:
Logging this can potentially be super expensive, we can log a hash of the script and that would be enough, as the full script has to be stored somewhere onchain anyway.

Also the `CallsScript` executor is logging all individual calls already.

Reply:
As far as I can tell with remix and the docs, logging more bytes doesn't actually cost more gas ATM. It's only indexed `bytes` and `string` parameters that get automatically keccaked (to provide the indexing).

From http://solidity.readthedocs.io/en/v0.4.24/contracts.html#events:

> If arrays (including string and bytes) are used as indexed arguments, the Keccak-256 hash of it is stored as topic instead.

----------------

Debating whether or not we should keccak / index these parameters anyway, for future-proofing.

Reply:
Each byte of data logged costs 8 gas as defined in the yellowpaper:
![2018-07-16 at 10 49 16 am](https://user-images.githubusercontent.com/447328/42750006-f80b8a50-88e5-11e8-9c24-62793f1f6be0.png)

However I gave this a second thought and I think it is not a bad idea to log scripts as some forwarders never put scripts on storage.

Reply:
Just retested this and confirmed that it does cost gas.